tests/flytekit/unit/core/test_shim_task.py

```diff
@@ -38,14 +38,14 @@ def test_resolver_load_task():
 
     resolver = TaskTemplateResolver()
     ts = get_serializable(OrderedDict(), serialization_settings, square)
-    with tempfile.NamedTemporaryFile() as f:
-        write_proto_to_file(ts.template.to_flyte_idl(), f.name)
```

Comment:
we will open the temporary file without closing it at lines 41 and 42 respectively, and it doesn't work on Windows.
```python
def write_proto_to_file(proto, path):
    Path(_os.path.dirname(path)).mkdir(parents=True, exist_ok=True)
    with open(path, "wb") as writer:
        writer.write(proto.SerializeToString())
```

Refer https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile